### PR TITLE
Improvements to interface documentation

### PR DESCRIPTION
**Pull request by mbabker**
This is an effort to improve the inline documentation in the package, starting with interfaces and a couple other base classes.  The biggest changes include:

- Review for grammar
- Adds types in doc blocks where there weren't any, and makes existing types more specific where able

### CODE REVIEW

**Review by franmomu — APPROVED** [review]
LGTM 🚀 after the conflict resolution

File: src/Loggable/Mapping/Event/LoggableAdapter.php [hunk 1]
```diff
@@ -2,36 +2,36 @@
...
  * @license MIT License (http://www.opensource.org/licenses/mit-license.php)
  */
 interface LoggableAdapter extends AdapterInterface
 {
     /**
-     * Get default LogEntry class used to store the logs
+     * Get the default object class name used to store the log entries.
      *
-     * @return string
+     * @return class-string
```
Comment (phansys) [inline src/Loggable/Mapping/Event/LoggableAdapter.php:19]: IMO, in order to keep the "native" docblocks clean, we should use the `phpstan-` prefix for cases like "class-string", "array-key", generics, etc.
```suggestion
     * @return string
     *
     * @phpstan-return class-string
```

File: src/Mapping/Event/AdapterInterface.php [hunk 2]
```diff
@@ -5,12 +5,11 @@
 use Doctrine\Common\EventArgs;
 use Doctrine\ORM\UnitOfWork;
```
Comment (phansys) [inline src/Mapping/Event/AdapterInterface.php:6]: IMO, if this class is not used, we should remove this import.

**Review by phansys — CHANGES_REQUESTED** [review]

File: src/Loggable/Mapping/Event/LoggableAdapter.php (hunk 1 above)
Comment (mbabker) [reply src/Loggable/Mapping/Event/LoggableAdapter.php]: Covered in the rebase

File: src/Mapping/Event/AdapterInterface.php (hunk 2 above)
Comment (mbabker) [reply src/Mapping/Event/AdapterInterface.php]: It's used.

**Review by phansys — APPROVED** [review]

**Comment by phansys** [comment]
Thank you @mbabker!

**Pull request merged**